src/models/vartype.ts

```diff
@@ -4,12 +4,33 @@ export interface VarTypeDictionary {
     [ slug: string ]: VarType;
 }
 
+export class SchemaKey {
+  type: string;
+  units: string;
+  decimal: number;
+  label: string;
+  outputUnits: any;
+
+  constructor(data: any = {}) {
+    this.type = data['type'];
+    this.units = data['units'];
+    this.decimal = data['decimal'];
+    this.label = data['label'];
+    this.outputUnits = data['output_units'];
+  }
+}
+
+export class Schema {
+  schemaKeys: Array<SchemaKey>;
```

Comment:
Sorry if I said to make it an array, but you should be questioning me if I say something that does not make sense. Or how were you planing to use this as an array?. 

Converting the dictionary to an array as you do below makes no sense. It should stay as a dictionary so it is easy to query.

It should be a dictionary using something like

```
export interface SchemaKeyDictionary {
    [ index: string ]: SchemaKey;
}
```